### PR TITLE
Openocd config file added to flash and debug with ftdi interface

### PR DESCRIPTION
**Pull request by g-lesssard**
Added a config file for openocd. Flashing via JTAG works, debugging works with VS Code. An example launch and task file has been provided in readme.

### CODE REVIEW

**Review by cquesnel — APPROVED** [review]

File: tools/openocd/adafruit-esp.cfg [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
```
Comment (xgroleau) [inline tools/openocd/adafruit-esp.cfg:1]: Why is adafruit in the filename?

File: README.md [hunk 2]
```diff
@@ -74,6 +74,108 @@ This has been tested on Ubuntu 20.04 but the same logic should apply to any syst
...
+                    "beginsPattern": ".",
+                    "endsPattern": ".",
+                  }
+                }
+              ],
+            "options": 
+            {
+                "cwd": "/home/casto/.espressif/tools/openocd-esp32/v0.10.0-esp32-20191114/openocd-esp32" // to be changed by user
+            },
+            "command": "bin/openocd -s share/openocd/scripts -f ${workspaceFolder}/tools/openocd/adafruit-esp.cfg -c 'program_esp ${workspaceFolder}/cmake-build-target/esp32-boilerplate.bin 0x10000'",
```
Comment (xgroleau) [inline README.md:159]: woulnd't it be more convenient to have this in the launch.json. The field openocd would be as the debugServerPath and use the debugServerArgs for the args. 

Then have a task.json as preLaunchTask that will build your project (so it calls cd $dir && make)  so your build is up to date?

**Review by xgroleau — COMMENTED** [review]
Just suggestions some. Take them or not, it's a boilerplate anyway so it's more as a reference and for experimenting

File: tools/openocd/adafruit-esp.cfg (hunk 1 above)
Comment (g-lesssard) [reply tools/openocd/adafruit-esp.cfg:1]: The interface we are using is the adafruit breakout board. A different sourcing of the ft2232h chip could have a different VID. In the future, if we have the same VID on the chip used on the Hiveboard, we could rename it to hiveboard-esp.cfg.

Comment (xgroleau) [reply tools/openocd/adafruit-esp.cfg:1]: Thanks for the clarification!

File: README.md (hunk 2 above)
Comment (g-lesssard) [reply README.md:160]: I had trouble getting the debugServer with vs code and documentation is hard to find on this type of configuration. I added a make command to the preLaunchTask so the build is up to date

**Pull request merged**